src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -948,20 +953,24 @@ protected async override Task<SearchResult> SearchForReindexInternalAsync(Search
                     if (results.Results.Any())
                     {
                         results.MaxResourceSurrogateId = results.Results.Max(e => e.Resource.ResourceSurrogateId);
-                        break;
+                        _logger.LogInformation("For Reindex, Resource Type={ResourceType} Count={Count} MaxResourceSurrogateId={MaxResourceSurrogateId}", resourceType, results.TotalCount, results.MaxResourceSurrogateId);
+                        return results;
                     }
 
                     _logger.LogInformation("For Reindex, empty data page encountered. Resource Type={ResourceType} StartId={StartId} EndId={EndId}", resourceType, range.StartId, range.EndId);
                 }
+
+                // If no resources are found in the group of surrogate id ranges, move forward the starting point.
+                if (ranges.Any())
```

Comment:
Another question :) based on the check below: `while (ranges is not null`

Does the nullability apply here too?
```suggestion
                if (ranges?.Any() == true)
```

Reply:
nullability doesn't apply to either, good catch. removed from the while statement. 👍🏽

Reply:
I think consistency, null is either ok or it isn't :)